homeassistant/components/unifiprotect/switch.py

```diff
@@ -0,0 +1,252 @@
+        entity_category=EntityCategory.CONFIG,
+        ufp_value="osd_settings.is_debug_enabled",
+        ufp_set_function="set_osd_bitrate",
+    ),
+    ProtectSwitchEntityDescription(
+        key=_KEY_SMART_PERSON,
+        name="Detections: Person",
+        icon="mdi:walk",
+        entity_category=EntityCategory.CONFIG,
+        ufp_required_field="feature_flags.has_smart_detect",
+        ufp_value="is_person_detection_on",
+        ufp_set_function="set_person_detection",
+    ),
+    ProtectSwitchEntityDescription(
+        key=_KEY_SMART_VEHICLE,
+        name="Detections: Vehicle",
+        icon="mdi:car",
+        entity_category=EntityCategory.CONFIG,
+        ufp_required_field="feature_flags.has_smart_detect",
+        ufp_value="is_vehicle_detection_on",
+        ufp_set_function="set_vehicle_detection",
+    ),
+)
+
+
+LIGHT_SWITCHES: tuple[ProtectSwitchEntityDescription, ...] = (
+    ProtectSwitchEntityDescription(
+        key=_KEY_STATUS_LIGHT,
+        name="Status Light On",
+        icon="mdi:led-on",
+        entity_category=EntityCategory.CONFIG,
+        ufp_value="light_device_settings.is_indicator_enabled",
+        ufp_set_function="set_status_light",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors for UniFi Protect integration."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[ProtectDeviceEntity] = async_all_device_entities(
+        data,
+        ProtectSwitch,
+        all_descs=ALL_DEVICES_SWITCHES,
+        camera_descs=CAMERA_SWITCHES,
+        light_descs=LIGHT_SWITCHES,
+    )
+    async_add_entities(entities)
+
+
+class ProtectSwitch(ProtectDeviceEntity, SwitchEntity):
+    """A UniFi Protect Switch."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        device: ProtectAdoptableDeviceModel,
+        description: ProtectSwitchEntityDescription,
+    ) -> None:
+        """Initialize an UniFi Protect Switch."""
+        self.entity_description: ProtectSwitchEntityDescription = description
+        super().__init__(data, device)
+        self._attr_name = f"{self.device.name} {self.entity_description.name}"
+        self._switch_type = self.entity_description.key
+
+        if not isinstance(self.device, Camera):
+            return
+
+        if self.device.is_privacy_on:
+            self._previous_mic_level = 100
+            self._previous_record_mode = RecordingMode.ALWAYS
+        else:
+            self._previous_mic_level = self.device.mic_volume
+            self._previous_record_mode = self.device.recording_settings.mode
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        assert self.entity_description.ufp_value is not None
+
+        ufp_value = get_nested_attr(self.device, self.entity_description.ufp_value)
+        if self._switch_type == _KEY_HIGH_FPS:
```

Comment:
We could add a value_fn to the entity description mix in and prefer that

Reply:
Never mind it's not worth it for only one key, we can do that if there are more later.